internal/controller/drplacementcontrolvolsync.go

```diff
@@ -13,44 +13,32 @@ import (
 	ctrlutil "sigs.k8s.io/controller-runtime/pkg/controller/controllerutil"
 )
 
-func (d *DRPCInstance) EnsureVolSyncReplicationSetup(homeCluster string) error {
-	d.log.Info(fmt.Sprintf("Ensure VolSync replication has been setup for cluster %s", homeCluster))
+func (d *DRPCInstance) EnsureSecondaryReplicationSetup(srcCluster string) error {
+	d.setProgression(rmn.ProgressionEnsuringVolSyncSetup) // TODO: Update progression string
 
-	if d.volSyncDisabled {
-		d.log.Info("VolSync is disabled")
-
-		return nil
-	}
-
-	vsRepNeeded, err := d.IsVolSyncReplicationRequired(homeCluster)
+	// Create or update the destination VRG
+	opResult, err := d.createOrUpdateSecondaryManifestWork(srcCluster)
 	if err != nil {
 		return err
 	}
 
-	if !vsRepNeeded {
-		d.log.Info("No PVCs found that require VolSync replication")
-
-		return nil
+	if opResult == ctrlutil.OperationResultCreated {
+		return WaitForVolSyncManifestWorkCreation
 	}
 
-	return d.ensureVolSyncSetup(homeCluster)
-}
-
-func (d *DRPCInstance) ensureVolSyncSetup(srcCluster string) error {
-	d.setProgression(rmn.ProgressionEnsuringVolSyncSetup)
+	if _, found := d.vrgs[srcCluster]; !found {
+		return fmt.Errorf("failed to find source VRG in cluster %s. VRGs %v", srcCluster, d.vrgs)
+	}
 
-	// Create or update the destination VRG
-	opResult, err := d.createOrUpdateVolSyncDestManifestWork(srcCluster)
+	vsRepNeeded, err := d.IsVolSyncReplicationRequired(srcCluster)
```

Comment:
I suggest extracting the code from this line onward in this function into a separate function, perhaps retaining the original name ensureVolSyncSetup. This way, the top portion remains shared for both (VolRep and VolSync), while everything from this line down is specific to VolSync.